matchconfig.fbs

```diff
@@ -0,0 +1,409 @@
+  respawn_time:RespawnTimeMutator;
+  /// Match max duration including kickoff, replays, and more.
+  // TODO: Is this correct? 
+  max_time:MaxTimeMutator;
+  /// Additional game haviour for custom modes.
+  game_event:GameEventMutator;
+  /// Additional audio options for custom modes.
+  audio:AudioMutator;
+}
+
+/// Possible behaviours when a match is started while another match is in progress.
+enum ExistingMatchBehavior : ubyte {
+  /// Always restart the match, even if config is identical.
+  Restart,
+  /// Never restart an existing match if possible, just try to remove or spawn cars to match the configuration.
+  /// If we are not in the middle of a match, a match will be started. Handy for LAN matches.
+  ContinueAndSpawn,
+  /// Restart the match if any match settings differ.
+  /// No other otherwise.
+  RestartIfDifferent,
+}
+
+/// Possible to launch Rocket League.
+enum Launcher : ubyte {
+  Steam,
+  Epic,
+  /// E.g. if you use Legendary.
+  /// The game path is specified in the MatchConfiguration.
+  Custom
+}
+
+/// A ScriptConfiguration defines a script of a match.
+table ScriptConfiguration {
+  name:string (required);
+  location:string (required);
+  run_command:string (required);
+  spawn_id:int;
+  agent_id:string (required);
+}
+
+/// Definition of a match.
+/// Can be sent to RLBot to request the start of a match.
+table MatchConfiguration {
+  /// How to launch Rocket League.
+  /// If left unset, RLBot will not launch the game.
+  launcher:Launcher;
+  /// The path to the main Rocket League binary.
+  /// Only used for launcher is set to Custom.
+  game_path:string (required);
+  /// If true, RLBot will start the bots with a non-empty run command in their player configuration.
+  auto_start_bots:bool;
+  /// The name of a upk file, like UtopiaStadium_P, which should be loaded.
+  /// On Steam version of Rocket League this can be used to load custom map files,
+  /// but on Epic version it only works on the Psyonix maps.
+  /// Available maps can be found here: https://github.com/VirxEC/python-interface/blob/master/rlbot/utils/maps.py
+  game_map_upk:string (required);
+  /// The players in the match.
+  player_configurations:[PlayerConfiguration] (required);
+  /// The custom scripts used in the match.
+  script_configurations:[ScriptConfiguration] (required);
+  /// The game mode.
+  game_mode:GameMode;
+  /// Whether to skip goal replays.
+  skip_replays:bool;
+  /// Whether to start without a kickoff countdown.
+  instant_start:bool;
+  /// Mutator settings.
+  mutators:MutatorSettings;
+  /// How to handle any ongoing match.
+  existing_match_behavior:ExistingMatchBehavior;
+  /// Whether debug rendering is displayed.
+  enable_rendering:bool;
+  /// Whether clients are allowed to manipulate the game state, e.g. teleporting cars and ball.
+  enable_state_setting:bool;
+  /// Whethre the match replay should be saved.
+  auto_save_replay:bool;
+  /// Whether the freeplay mode should be used.
+  // TODO: More details
```

Comment:
Not sure what other details can be added - RLBot works fine in free play! This will launch a free play match instead of an exhibition match. This allows the players to use training keybinds, bakkes plugins and whatnot that are allowed when you're in free play.

Reply:
That is exactly the kind of details I wanted to add :D